packages/runtime-core/src/apiOptions.ts

```diff
@@ -218,6 +244,9 @@ export function applyOptions(
   } = options
 
   const globalMixins = instance.appContext.mixins
+  // used to validate repeatedly declared property
+  const optionCache: Record<string, any> = Object.create(null)
+  const cachingOptions = cachingComponentOptions.bind(null, optionCache)
```

Comment:
1. I think the `cachingOptions` and `cachingComponentOptions ` naming is a bit confusing. The purpose of the function is to check duplicate properties, and caching is just an implementation detail.

2. Using bind here seems unnecessary. Here's what I'd do:

``` js
function createDuplicateChecker() {
  const cache = Object.create(null)
  return (type, key) => {
    // ...
  }
}

// also only call it only during dev
const checkDuplicateProperties = __DEV__ ? createDuplicateChecker() : null
```

Reply:
thank you for your suggestion, it's very useful.